packages/firebase_performance/lib/src/firebase_performance.dart

```diff
@@ -13,64 +13,70 @@ enum HttpMethod { Connect, Delete, Get, Head, Options, Patch, Post, Put, Trace }
 class FirebasePerformance {
   FirebasePerformance._();
 
-  static int _traceCount = 0;
-  static int _httpMetricCount = 0;
-
   @visibleForTesting
   static const MethodChannel channel =
       MethodChannel('plugins.flutter.io/firebase_performance');
 
   /// Singleton of [FirebasePerformance].
   static final FirebasePerformance instance = FirebasePerformance._();
 
+  static int _channelCount = 0;
+
   /// Determines whether performance monitoring is enabled or disabled.
   ///
   /// True if performance monitoring is enabled and false if performance
   /// monitoring is disabled. This is for dynamic enable/disable state. This
   /// does not reflect whether instrumentation is enabled/disabled.
-  Future<bool> isPerformanceCollectionEnabled() async {
-    final bool isEnabled = await channel
-        // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-        // https://github.com/flutter/flutter/issues/26431
-        // ignore: strong_mode_implicit_dynamic_method
-        .invokeMethod('FirebasePerformance#isPerformanceCollectionEnabled');
-    return isEnabled;
+  Future<bool> isPerformanceCollectionEnabled() {
+    return channel.invokeMethod<bool>(
+      'FirebasePerformance#isPerformanceCollectionEnabled',
+    );
   }
 
-  /// Enables or disables performance monitoring.
+  /// Enables or disables performance monitoring asynchronously.
```

Comment:
I feel like asynchronously is strongly implied by the method signature and may not be necessary to call out explicitly here. Most plugin methods that don't return a value *should* be `Future<void>`, and this one isn't unusual.